### ISSUE TITLE
'NoneType' object has no attribute 'sysadmin'

### ISSUE THREAD

**Issue by napulen** [frictionlessdata/ckanext-validation#35]
Hi, I am getting this error when trying to run synchronous validation while I am creating a csv resource

By running in debug mode, I get this output when trying to add the resource
![image](https://user-images.githubusercontent.com/7258463/38646091-9de9c770-3db4-11e8-9be6-18a6316724c5.png)

It seems that it is crashing here:

/usr/lib/ckan/default/src/ckan/ckan/logic/validators.py

The action=owner_org_validator is getting a null user object for some reason.

Any ideas?

**Comment by amercader** [comment]
@napulen thanks for the report. Can you paste the full stack trace somewhere? Are you using any custom [authorization settings](http://docs.ckan.org/en/latest/maintaining/configuration.html#authorization-settings) or just the default ones? Do you customize the authorization from an extension?

I assume you are logged in when you try to create the dataset with the CSV right?

**Comment by napulen** [comment]
@amercader sorry, it is difficult for me to reach the full stack at this moment, I do not think we are using any custom authorization settings or customizing the authorization from an extension... Though I am not completely sure...

However, I have a better idea of what was going on now and managed to get through this issue...

I was reading your code and you are overwriting the ```resource_create``` action

```
def resource_create(context, data_dict):
    '''Appends a new resource to a datasets list of resources.

    This is duplicate of the CKAN core resource_create action, with just the
    addition of a synchronous data validation step.

    This is of course not ideal but it's the only way right now to hook
    reliably into the creation process without overcomplicating things.
    Hopefully future versions of CKAN will incorporate more flexible hook
    points that will allow a better approach.
```
The csv file that I was submitting had errors, so the branch of the code that had to be executed was the one in charge of dealing with invalid files. At some point, this ends up in a call to a "resource_delete" action

```
    if not report['valid']:   #<---- THE REPORT IS NOT VALID

        # Delete validation object
        t.get_action(u'resource_validation_delete')(
            {u'ignore_auth': True},
            {u'resource_id': resource_id}
        )

        # Delete uploaded file
        if upload:
            delete_local_uploaded_file(resource_id)

        if new_resource:
            # Delete resource
            t.get_action(u'resource_delete')(  # <-- DELETE THE RESOURCE
                {u'ignore_auth': True, 'user': None},
                {u'id': resource_id}
            )
```
The error I am getting from the ckan website is that a validator plugin is crashing when looking for the attributes of a "user" object. It seems this originates from the call to the `resource_delete` action. You can see that the call has an explicit null user object.

So, we went one level above (the wrapper for resource_create), took the user from the context and sent it to the `_run_sync_validation` function, and when we reach the call to the `resource_delete` action, we are adding the user that was received.

I don't know if that is a real problem in the extension or we introduced somehow, but that worked, after those changes we have been able to run the validation.

The code looks like this: 

```
        if new_resource:
            # Delete resource
            t.get_action(u'resource_delete')(
                {u'ignore_auth': True, 'user': user},
                {u'id': resource_id}
            )
```
I can submit a pull request with the change if needed.
 
Thanks

**Comment by amercader** [comment]
Thanks @napulen. The problem is that the `owner_org_validator` does not honour the `ignore_auth` flag and tries to look for a user. This was fixed in 2.8 as part of a [larger change](https://github.com/ckan/ckan/commit/c15814ee260416b6401c188e719d26cd9003c7a5#diff-8525268a0dc8fbd61bf4f7c4cda26389), and we should backport it. In the meantime the workaround you suggest looks good.

**Comment by amercader** [comment]
This should be fixed if people uses the latest patch releases (2.7.4, 2.6.6, 2.5.9 or higher)
